### PR TITLE
feat: update to Angular 19

### PR DESCRIPTION
**Pull request by mshima**
Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) and [DEVELOPER.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/DEVELOPER.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

**Comment by maxokorokov** [comment]
Fixes #4779

File: demo/src/main.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ bootstrapApplication(AppComponent, {
 		provideHttpClient(withNoXsrfProtection()),
 		provideExperimentalZonelessChangeDetection(),
 		{
```
Comment (quentinderoubaix) [inline demo/src/main.ts:32]: No need for `useFactory` anymore, you can refactor like this:
```ts
provideExperimentalZonelessChangeDetection(),
provideAppInitializer(() => inject(AnalyticsService).start())
```

**Review by quentinderoubaix — COMMENTED** [review]
Thanks for the contribution @mshima !
Seems good to me over all.

Left-over stuff includes:
- [x] update peer dependencies in **src/package.json**
- [x] update the **getting-started.page.html**

In another PR, we can have a little review to remove the warnings of unused imports

**Review by quentinderoubaix — APPROVED** [review]
LGTM !

**Pull request merged**